content/blog/gain-platform-superpowers-with-the-kebap-stack/index.md

```diff
@@ -0,0 +1,704 @@
+
+  resources:
+    do_cluster:
+      type: digitalocean:KubernetesCluster
+      properties:
+        name: "${clusterName}"
+        region: "${clusterRegion}"
+        version: "${version}"
+        destroyAllAssociatedResources: true
+        nodePool:
+          name: "${nodePoolName}"
+          size: "s-2vcpu-2gb"
+          nodeCount: "${nodeCount}"
+    # cut for brevity
+---
+apiVersion: pulumi.com/v1
+kind: Stack
+metadata:
+  name: ${{values.name}}
+spec:
+  stack: ${{values.stack}}
+  envRefs:
+    DIGITALOCEAN_TOKEN:
+      type: Env
+      env:
+        name: DIGITALOCEAN_TOKEN
+  programRef:
+    name: ${{values.name}}
+  destroyOnFinalize: false
+```
+
+### External Secrets Operator (ESO)
+
+<img src="e.png" >
+
+ESO addresses the challenge of securely managing secrets in Kubernetes. It integrates with various secret stores (such
+as AWS Secrets Manager, HashiCorp Vault, Google Secrets Manager, Azure Key Vault) in a consistent way. This allows you
+to configure access to the secret store once for all your required secrets, enhancing security and simplifying
+management.
+
+We even take it a step further by using ESO to manage the secrets from Pulumi ESC (External Secret Configurations) with
+the [Pulumi Provider](https://external-secrets.io/latest/provider/pulumi/) for ESO. This way, we can manage the secrets
+in the same way as the rest of the infrastructure, ensuring that the secrets are managed in a secure and consistent
+manner across different environments.
+
+To give the Pulumi Kubernetes Operator access to the secrets, we create following `ClusterSecretStore` object:
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: ClusterSecretStore
+metadata:
+  name: secret-store
+spec:
+  provider:
+    pulumi:
+      organization: <org>
+      environment: <env>
+      accessToken:
+        secretRef:
+          name: pulumi-access-token
+          key: PULUMI_ACCESS_TOKEN
+          namespace: argocd
+```
+
+And then we can define the `ExternalSecret` with all the secrets that we want to manage and pass to our Operator as
+environment variables `extraEnv`:
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: pulumi-operator-secrets
+spec:
+  refreshInterval: 20s
+  secretStoreRef:
+    kind: ClusterSecretStore
+    name: secret-store
+  data:
+  - secretKey: pulumi-access-token
+    remoteRef:
+      key: "backstage.pulumi-pat"
+  - secretKey: do-token
+    remoteRef:
+      key: "backstage.do"
+```
+
+### Backstage
+
+<img src="b.png" >
+
+Backstage is an open platform for building developer portals. It provides a centralized place for managing software
+catalogs, documentation, and tooling. This layer helps in organizing microservices and infrastructure, streamlining the
+process for developers to create, manage, and explore services.
+
+We also installed the [Pulumi plugin](https://github.com/pulumi/pulumi-backstage-plugin) for Backstage, which allows us
```

Comment:
Ok, I think I am understanding why some of this is reading funny. I am going to make this comment here, but I think it applies to how you explain Pulumi + GitOps or Pulumi + K8s operator. 

You are describing the technical implementation detail of things. In this case, using the plugin for Backstage. I actually don't care about the implementation. Readers want to hear about the workflow that is happening. For example..."Backstage provides a developer portal UI that sits atop of collections of Pulumi infrastructure as code program templates. Backstage eseentially is where the end user interacts with this platform. They select their template and Backstage through a Plugin calls Pulumi to provision and manage the infrastructure by executing the IaC program" or something like that. 

Describe the workflow of how the pieces work together. Don't describe just the technical implementation details.